src/test/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookConsumerTest.java

```diff
@@ -247,6 +251,36 @@ public void testShouldNotTriggerBuildIfRepositoryDoesNotMatch() {
         return singletonList(remoteConfig);
     }
 
+    @Test
+    public void testRefsChangedNotBitbucketSCM() {
+        GitSCMSource scmSource = mock(GitSCMSource.class);
+        BitbucketWebhookConsumer.BitbucketSCMHeadEvent headEvent = new BitbucketWebhookConsumer.BitbucketSCMHeadEvent(null, null, null);
+
+        assertThat(headEvent.heads(scmSource), equalTo(emptyMap()));
+    }
+
+    @Test
+    public void testRefsChangedNotMatchingMultibranchRepo() {
+        BitbucketSCMRepository mockScmRepo = mock(BitbucketSCMRepository.class);
+        doReturn("PROJ_1").when(mockScmRepo).getProjectKey();
+        doReturn("rep_1").when(mockScmRepo).getRepositorySlug();
+
+        BitbucketProject mockWebhookProject = mock(BitbucketProject.class);
+        BitbucketRepository mockWebhookRepo = mock(BitbucketRepository.class);
+        doReturn(mockWebhookProject).when(mockWebhookRepo).getProject();
+        doReturn("PROJ_2").when(mockWebhookProject).getKey();
+        doReturn("rep_2").when(mockWebhookRepo).getSlug();
+
+        BitbucketSCMSource scmSource = mock(BitbucketSCMSource.class);
+        RefsChangedWebhookEvent payload = mock(RefsChangedWebhookEvent.class);
+        doReturn(mockScmRepo).when(scmSource).getBitbucketSCMRepository();
+        doReturn(mockWebhookRepo).when(payload).getRepository();
+
+        BitbucketWebhookConsumer.BitbucketSCMHeadEvent headEvent = new BitbucketWebhookConsumer.BitbucketSCMHeadEvent(null, payload, null);
+
+        assertThat(headEvent.heads(scmSource), equalTo(emptyMap()));
+    }
+
```

Comment:
Maybe we should have a test checks that there *is* something in the map when it should?

Reply:
Do you think that case is already covered by the other unit tests here? It's a gnarly one to write it's own test, and I figured the "heads exist" case is covered (abeit not in a unit) by the other cases.

Reply:
I guess..